src/clj/game/core/commands.clj

```diff
@@ -40,7 +40,7 @@
   (set-prop state side target :advance-counter value)
   (system-msg state side (str "sets advancement counters to " value " on "
                               (card-str state target)))
-  (trigger-event state side :advancement-placed target))
+  (wait-for (trigger-event-simult state side (make-eid state) :advancement-placed nil target)))
```

Comment:
Should be left as `trigger-event`. If `set-adv-counter` isn't async, this will make things worse.